cmd/cmrel/cmd/gcb_publish.go

```diff
@@ -493,11 +496,24 @@ func pushContainerImages(ctx context.Context, o *gcbPublishOptions, rel *release
 	for name, tars := range rel.ComponentImageBundles {
 		log.Printf("Pushing release images for component %q", name)
 		for _, t := range tars {
-			if err := docker.Push(ctx, t.ImageName()); err != nil {
+			imageTag := buildImageTag(o.PublishedImageRepository, name, t.Architecture(), rel.ReleaseVersion)
+
+			log.Printf("Tagging %q with new name %q", t.RawImageName(), imageTag)
+
+			if err := docker.Tag(ctx, t.RawImageName(), imageTag); err != nil {
+				return err
+			}
+
+			if err := docker.Push(ctx, imageTag); err != nil {
 				return err
 			}
-			log.Printf("Pushed release image %q", t.ImageName())
-			pushedContent = append(pushedContent, t.ImageName())
+
+			// PublishedTag will be used later to refer to the image under the tag we
+			// actually pushed it under
+			t.PublishedTag = imageTag
```

Comment:
note: I'm not crazy about storing this tag on `t`, but it works well and is very simple. Some of the logic when processing a release gets a little hairy and this was an easy way of achieving this task!